core/commands/filestore.go

```diff
@@ -37,58 +48,89 @@ The output is:
 <hash> <size> <path> <offset>
 `,
 	},
-	Arguments: []cmds.Argument{
-		cmds.StringArg("obj", false, true, "Cid of objects to list."),
+	Arguments: []cmdsutil.Argument{
+		cmdsutil.StringArg("obj", false, true, "Cid of objects to list."),
 	},
-	Run: func(req cmds.Request, res cmds.Response) {
-		_, fs, err := getFilestore(req)
+	Run: func(req cmds.Request, re cmds.ResponseEmitter) {
+		_, fs, err := getFilestore(req.InvocContext())
 		if err != nil {
-			res.SetError(err, cmds.ErrNormal)
+			re.SetError(err, cmdsutil.ErrNormal)
 			return
 		}
 		args := req.Arguments()
 		if len(args) > 0 {
 			out := perKeyActionToChan(args, func(c *cid.Cid) *filestore.ListRes {
 				return filestore.List(fs, c)
 			}, req.Context())
-			res.SetOutput(out)
+
+			for v := range out {
+				re.Emit(v)
+			}
 		} else {
 			next, err := filestore.ListAll(fs)
 			if err != nil {
-				res.SetError(err, cmds.ErrNormal)
+				re.SetError(err, cmdsutil.ErrNormal)
 				return
 			}
+
 			out := listResToChan(next, req.Context())
-			res.SetOutput(out)
+			for v := range out {
```

Comment:
maybe since we do this often, we should have some sort of `re.PipeOutput(ch)` method